source/glimmer-components/passing-data-to-components.md

```diff
@@ -0,0 +1,231 @@
+Imagine if functions aren't allowed to take arguments: they won't be very
+useful at all, right? Likewise, if we cannot pass data to our Components, they
+would just be fancy partials with limited uses. Fortunately, Ember made it very
+easy to pass data into our Components.
+
+## Passing Attributes
+
+When invoking Components, you can pass along any number of named attributes,
+just like how you can set attributes on regular HTML elements:
+
+```app/templates/contact-us.hbs
+<h5><x-icon type="speech-bubble" size="large" /> Contact Us</h5>
+
+<x-icon type="phone" /> 1-800-EMBER-JS <br>
+<x-icon type="twitter" /> @emberjs <br>
+<x-icon type="map" /> 123 Some Street, Portland OR
+```
+
+These attributes can then be accessed from the Component's template via the
+`attrs` hash:
+
+```app/templates/components/x-icon.hbs
+<x-icon class="icon icon-{{attrs.type}} {{attrs.size}}" />
+```
+
+You might be wondering: what's the point of going through the `attrs` hash?
+Couldn't Ember just expose the attributes as `{{type}}` instead of the nosier
+`{{attrs.type}}`? Hold your thoughts! We will come back to that soon.
+
+In addition to string attributes, sometimes you might need to pass other kind
+of data such as numbers and booleans. You can do this using the curlies syntax
+instead of quotation marks:
+
+```app/templates/step-two.hbs
+<progress-bar value={{2}} max={{10}} animated={{true}} /> Step 2 / 10
```

Comment:
Because `<progress-bar value=2></progress-bar>` is already defined in HTML as meaning:

```html
<progress-bar value="2"></progress-bar>
```

We can get away with some divergence, like adding support for `<progress-bar />`, in areas where the thing we're adding is very, very unused in practice. In the case of `<progress-bar />`, it is actually far more likely that the HTML definition ("ignore the `/`") is not what the user intended.

However, when someone writes `<progress-bar value=2>` in an HTML-like context, there is a very strong likelihood that they really do mean the HTML semantics, so diverging here is not acceptable.